### ISSUE TITLE
Use Hot Code Reloading?

### ISSUE THREAD

**Issue by 0atman** [inim-repl/INim#69]
Like https://youtu.be/7WgCt0Wooeo

**Comment by Tangdongle** [comment]
Yeah, super keen for that. When I last had a look it only supported the JS backend, but now we can do much much more

**Comment by 0atman** [comment]
Works on the C backend too, apparently! https://nim-lang.org/docs/hcr.html

**Comment by gavr123456789** [comment]
oh, using a repl with hcr, you can create something similar to the Pharo / Glamorous Toolkit (Smalltalk) experience.

**Comment by 0atman** [comment]
@gavr123456789 oh amazing! There are Nim bindings for Jupyter notebooks, try https://github.com/stisa/jupyternim which HAS hot code reloading! Very impressive, we could learn a lot from this project.

**Comment by gavr123456789** [comment]
yes, it is difficult to imagine something similar to Pharo can be implemented in a statically typed compiled language.  
By the way, Nim and Smalltalk have a lot in common, oddly enough.  
Here is an interesting series of articles:  
http://goran.krampe.se/2014/10/29/nim-and-oo/  
http://goran.krampe.se/2014/10/31/nim-and-oo-part-ii/  
http://goran.krampe.se/2014/10/31/nim-and-oo-part-iii/  
http://goran.krampe.se/2014/11/03/squeak-to-nim/  
http://goran.krampe.se/2014/11/30/nim-and-oo-part-iv/
